### ISSUE TITLE
Message templates

### ISSUE THREAD

**Issue by AnthonySteele** [justeat/NLog.StructuredLogging.Json#83]
Open thread about pros and cons of message templates.
See https://messagetemplates.org/

So far we have not used message templates since we have not felt that they are ideal: the message is presented twice wasting bytes, and the string formatting used to render the "human-readable" version takes time.  When logging frequently in high throughput high performance systems, you don't want to squander these.

However,  message templates might be an emerging standard and therefore useful for logging consistently with other logging libraries.

**Comment by AnthonySteele** [comment]
Message Templates is now supported widely, e.g. `Serilog`,  `NLog` and by `
Microsoft.Extensions.Logging.Abstractions.ILogger`.

We should at least know how we interact with it.

**Comment by AnthonySteele** [comment]
See also #124

**Comment by AnthonySteele** [comment]
Code to make it work well in PR #126 
tl;dr: with this, you can use [the standard NLog implementation of `ILogger`](https://github.com/NLog/NLog.Extensions.Logging) and templated messages, and it should all work.
